auth/api/iam/s2s_vptoken.go

```diff
@@ -115,20 +141,19 @@ func (r Wrapper) RequestAccessToken(ctx context.Context, request RequestAccessTo
 	return RequestAccessToken200JSONResponse(*tokenResult), nil
 }
 
-func (r Wrapper) createAccessToken(issuer did.DID, issueTime time.Time, presentation vc.VerifiablePresentation, scope string) (*oauth.TokenResponse, error) {
+func (r *Wrapper) createAccessToken(issuer did.DID, issueTime time.Time, presentations []vc.VerifiablePresentation,
+	submission pe.PresentationSubmission, definition PresentationDefinition, scope string) (*oauth.TokenResponse, error) {
 	accessToken := AccessToken{
 		Token:  crypto.GenerateNonce(),
 		Issuer: issuer.String(),
 		// TODO: set ClientId
-		ClientId:   "",
-		IssuedAt:   issueTime,
-		Expiration: issueTime.Add(accessTokenValidity),
-		Scope:      scope,
-		// TODO: set values
-		InputDescriptorConstraintIdMap: nil,
-		VPToken:                        []VerifiablePresentation{presentation},
-		PresentationDefinition:         nil,
-		PresentationSubmission:         nil,
+		ClientId:               "",
```

Comment:
It is currently not issued to a specific DID, as there might be multiple credential subject DIDs in the `vp_token`?
I think `client_id` should reflect the authenticated client, but the client doesn't provide a client_id (and certainly not an authenticated client_id).

The authorization code grant's JAR (signed authorization request object), but we don't have that concept here. Ideas;
- specify client_id authentication based on signature with key from DID document
- authenticate client_id using client certificate, which must resolve to did:web being used as client_id
- ...?

Reply:
RFC021 states the following for the introspection:

> The client (DID) that requested the access token.

In the request, the `sub` field indicates the requester. So the DID from that field should be transfered to the client_id.

Reply:
`sub` == `iss` of the token == resource owner. Sounds like it's missing from the spec. Tokens should be issued to a specific party, and we should be able to check if that party is trusted. 

For now we expect a single VP in the s2s flow and all credentials in a VP must have the same subject == `client_id`?

Reply:
`sub` == `iss` of the token == resource owner. Sounds like it's missing from the spec. Tokens should be issued to a specific party, and we should be able to check if that party is trusted. 

For now we expect a single VP in the s2s flow and all credentials in a VP must have the same subject == `client_id`?

Reply:
In our context I think the `client_id` should always be a DID. In OpenID4VP the `client_id_scheme ` can be used to specify a DID as `client_id` in the Authorization Request. Not sure how to use that information, just thought it could be useful in this discussion.

maybe create a new issue and point to it in a TODO for now? IMO it's a blocking issue for production if we can't assign the token to a specific DID.

Reply:
Just took first VP signer for now, needs to be addressed in RFC first (https://github.com/nuts-foundation/nuts-specification/issues/269)